internal/ingress/controller/parser/parser.go

```diff
@@ -590,6 +626,26 @@ func overrideRoute(route *Route,
 	}
 }
 
+func overrideRoute(route *Route,
+	kongIngress *configurationv1.KongIngress) {
+	if route == nil {
+		return
+	}
+
+	if kongIngress != nil && kongIngress.Route != nil {
+		overrideRouteByKongIngress(route, kongIngress)
+	}
+
+	routeAnns := route.Ingress.Annotations["configuration.konghq.com/protocols"]
+
+	if len(routeAnns) > 0 && routeAnns == "grpc,grpcs" {
+		// TODO: don't hardcode the things
+		route.Protocols = kong.StringSlice("grpc", "grpcs")
+		// grpc(s) doesn't accept strip_path
+		route.StripPath = kong.Bool(false)
```

Comment:
You might be better off by refactoring the code as:
- first override by kongingress
- second, override using the annotation
- finally, see the values of protocol, and set the necessary properties to `nil` (StripPath for route, and Path for service)

This simplifies the code by getting rid of a lot of `if`s.